src/imgui/ImGuiMachine.cc

```diff
@@ -279,6 +267,99 @@ void ImGuiMachine::showMenu(MSXMotherBoard* motherBoard)
 			});
 		}
 
+		setupSettingsOpen = im::Menu("Setup settings", true, [&]{
+			ImGui::TextUnformatted("When openMSX starts:");
+			ImGui::RadioButton("Load a machine", &startType, StartType::MACHINE);
+			HelpMarker("Select this option to load the specified machine when openMSX starts up, instead of a setup.");
+			im::Indent([&] {
+				im::Disabled(startType != StartType::MACHINE, [&] {
+					auto& allMachines = getAllMachines();
+					auto currentDefault = std::string(reactor.getDefaultMachineSetting().getString());
+					auto currentDefaultDisplay = [&] {
+						if (auto* info = findMachineInfo(currentDefault)) {
+							return info->displayName;
+						}
+						return currentDefault;
+					}();
+					im::Combo("##defaultMachine", currentDefaultDisplay.c_str(), [&]{
+						for (auto& info: allMachines) {
+							bool ok = getTestResult(info).empty();
+							im::StyleColor(!ok, ImGuiCol_Text, getColor(imColor::ERROR), [&]{
+								if (ImGui::Selectable(info.displayName.c_str())) {
+									reactor.getDefaultMachineSetting().setValue(TclObject(info.configName));
+									// clear the default setup setting to avoid loading that at startup
+									reactor.getDefaultSetupSetting().setValue(TclObject(""));
+								}
+								if (info.configName == currentDefault && ImGui::IsWindowAppearing()) ImGui::SetScrollHereY();
+								if (ImGui::IsItemHovered(ImGuiHoveredFlags_DelayNormal | ImGuiHoveredFlags_NoSharedDelay | ImGuiHoveredFlags_Stationary)) {
+									im::ItemTooltip([&]{
+										printConfigInfo(info);
+									});
+								}
+							});
+						}
+					});
+				});
+			});
+			ImGui::RadioButton("Load a setup", &startType, StartType::SETUP);
+			HelpMarker("Select this option to load the specified setup when openMSX starts up. "
+				"Note that in case the setup cannot be loaded, the machine shown at the other "
+				"option will be loaded after all.");
+			im::Indent([&] {
+				im::Disabled(startType != StartType::SETUP, [&] {
+					if (!setupSettingsOpen) {
+						setups = Reactor::getSetups();
+					}
+					auto currentDefault = std::string(reactor.getDefaultSetupSetting().getString());
+					/* TODO: debug and improve this:
+					auto currentSaveAtExitName = manager.getReactor().getSaveSetupAtExitNameSetting().getString();
+					if (!contains(setups, currentSaveAtExitName)) {
+						setups.emplace_back(currentSaveAtExitName);
+					}
+					*/
+					im::Combo("##defaultSetup", currentDefault.c_str(), [&]{
+						for (auto& setup: setups) {
```

Comment:
We should also allow to select the "save name", even if there exist no such corresponding `.oms` file yet.

We cannot show a preview in that case. And maybe it's better to not indicate that as an error when we try to show the preview anyway?

Reply:
Addressed this, please check.